### PR TITLE
[FABCN-422] Add release guide

### PR DESCRIPTION
**Pull request by jt-nti**
Signed-off-by: James Taylor <jamest@uk.ibm.com>

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+- docker images
+    - [fabric-nodeenv](https://hub.docker.com/r/hyperledger/fabric-nodeenv)
+- npm modules
+    - [fabric-contract-api](https://www.npmjs.com/package/fabric-contract-api)
+    - [fabric-shim](https://www.npmjs.com/package/fabric-shim)
+    - [fabric-shim-api](https://www.npmjs.com/package/fabric-shim-api)
+    - [fabric-shim-crypto](https://www.npmjs.com/package/fabric-shim-crypto)
+
+**Note:** A docker image with a matching V.R version is required before releasing a new version of Fabric.
```
Comment (mbwhite) [inline RELEASING.md:13]: V2 only statement...

File: RELEASING.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+## Before releasing
+
+It's useful to create an issue to keep track of each release, for example [FABCN-377 Release v2.0.0 chaincode-node](https://jira.hyperledger.org/browse/FABCN-377).
+
+The following tasks are normally required before releasing, in addition to any specific to a particular release:
```
Comment (mbwhite) [inline RELEASING.md:19]: what are the conditions for 'normally' to not apply?

File: RELEASING.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+- Create a new release notes file
+- Update the `CHANGELOG.md` file
+  
+  The `changelog.sh` script in `tools/scripts` will prepopulate the changelog but you must check and edit the file manually afterwards as required
+
+See the [Prepare 2.1.4 release](https://github.com/hyperledger/fabric-chaincode-node/pull/174) pull request for an example, although be careful to search for all versions in the codebase as they're easy to miss and things change!
+
+## Create release
+
+Creating a GitHub release on the [releases page](https://github.com/hyperledger/fabric-chaincode-node/releases) will trigger the build to publish the new release.
```
Comment (mbwhite) [inline RELEASING.md:33]: Might want to include some comment about the name of the tag and what to put in the text.

File: RELEASING.md (hunk 1 above)
Comment (jt-nti) [reply RELEASING.md:13]: Although it's not used for releases, I'm assuming master is the same. Is it V.R.M for when I add the release process to the 1.4 branch, or is the docker image not required? Is there even a docker image?

Comment (mbwhite) [reply RELEASING.md:13]: For 1.4 it uses the `ccenv` rather than have it's own dedicated one.

**Pull request merged**